### DISCUSSION TITLE
New v3 firmware starting to roll out

### DISCUSSION THREAD

**Discussion by skilau** [gtxaspec/wz_mini_hacks#355] [General]
https://www.reddit.com/r/wyzecam/comments/y7n5x9/wyze_cam_v3_436102700_and_wyze_cam_pan_v2/

> We're starting a gradual release for Wyze Cam v3 4.36.10.2700 and Wyze Cam Pan v2 4.49.3.2705 firmware!

I wonder if they broke us, like they did with the v2 new firmware.
Anyone try it yet?

**Comment by gtxaspec** [comment]
they did, all new firmware releases for all devices are broken.  I'll update the readme today.  you can use wz_mini to get shell access to the device, but thats it.  nothing else works.

**Reply by skilau** [reply]
@gtxaspec : Thanks for confirming what I suspected.

Mine still show no update available.

Do you know if they are forcing the new firmware to the device?
Or is it still like before, where we have to select and allow them to do it?

PS: Notice this line too:

>Due to the security improvement on this firmware, you will need to go through the 
> factory reset process for your camera if you go back to firmware 4.36.9.139 or below

Which is the current firmware.   So they are going to make it very difficult if someone upgrades to the new version.

I probably should change my config to set DISABLE_FW_UPGRADE to true, just in case they try something tricky.

**Reply by gtxaspec** [reply]
the reset is required, due to the new way they store the wifi ssid and wifi pw.  They are encrypted now, so if you try to downgrade, the credentials are encrypted, which the old fw does not recognize.

**Reply by skilau** [reply]
I have not downloaded the new firmware yet, but clearly they did this to try to stop us from using this stuff.

Obviously we can stay on the older firmware, but over time, they will probably kick cams off that are on the older firmware.

I may try downloading it tonight, and checking out how bad things are, and possible ways we could work around it.

You said we can still get a shell?
Is that over ssh? telnet?  serial lines?
(If ssh, does it use our cert/key? Or some default user/password combination?)

Thanks!

**Reply by sideup66** [reply]
@skilau for the main shell youll be able to do that over SSH using the wz_mini hack. Im imagining crond would work as well still as that simply relies on a shell script to launch it....im curious if the dnsmasq fix is still present and working however. If that is working as well, the update means nothing in my personal setup as what im using from it simply relies on that (my setup is pretty basic honestly). 

As for cmd, they mean remote camera controls and RTSP functionality. Main reason Wyze is hunting the RTSP functionality is due to it cutting into one of their paid models. I believe camplus gives you SOME web based RTSP functionality from a browser as well as a firmware to kinda appease folks. Either way, Ive seen a previous merge that involved a patched icamera binary. Im thinking this may be the way to get back to RTSP and web based control. Maybe we can create a toggle in the conf file for loading the custom icamera binary, which can unlock all of the non shell based options. Wouldnt be a hard change honestly....

**Reply by Bob-AOL** [reply]
A toggle has just recently been introduced that puts the camera into 100% non-tethered mode (from Wyze) and back again (if desired). Check toward the end of #38 for the most recent developments.

**Reply by sideup66** [reply]
Sure, but this wasnt verified (as far as im aware with the latest update). even with that feature enabled you lose any libcallback functionality or CMD events. This means your camera cannot perform any of its normal functions in offline mode using this method. The new icamera binary integrates all external libraries into itself and I read that also they actively block certain things from the SD card now... since we still have the custom kernel + busybox root, we should be able to do more id think. But patching binaries is outside my wheelhouse, unfortunately :(.

**Reply by Bob-AOL** [reply]
Link to the merge you cite?

**Reply by sideup66** [reply]
That wasnt a merge, it was the change that wyze performed in the latest firmware update. There's a new version of icamera present that merges all libraries into it.

**Reply by ColtonYYZ** [reply]
This has probably already been said but I am thinking of asking my friend who has family in Japan, order some of the identical v3's and this new cam that gives you controls.. https://www.atomtech.co.jp/products/atomcamswing

*Please Note; I will gladly edit this reply if urls are not allowed. I mean no malice nor am I a sales person. I am just getting tired of Wyze and all their shenanigans!

**Comment by sideup66** [comment]
The part that frankly concerns me on this front, is that the way wyze seems to be acting, I get the impression they are going to try to force everyone to the new firmware or deliberately break compatibility with the app under the new firmware version. The argument will be the same as always "safety and security".... Not sure if any work is being done by @gtxaspec behind the scenes, but would love to try to help if needed.

**Comment by endertable** [comment]
The latest V2 firmware is being put out as a "FORCED" update, meaning eventually it will be pushed out automatically to all online cams.  I imagine the other cams will follow suit with their updates.  From a devil's advocate perspective, Wyze needs to tighten up on the security of their cams as they are always getting called out for that and from a business model they would be smart to shut down anything that competes with their money making services.   :(

With that being said, for those that are offline, it's not a big deal and there really is no major reason to try to hack the latest firmwares anyway if you have a working CCTV system with frigate or some other NVR system.  They don't usually improve the video or network systems, most of their updates are for security or CamPlus features and other minor fixes.  There is no doubt in my mind Wyze has eyes on wyze_mini (as many users reference it even on the Wyze forums) so any circumventing/hacking mentioned here they can put counter-action on, as what happens with any hack that gets "too big".

**Reply by Bob-AOL** [reply]
@gtxaspec So is backward compatibility to earlier firmware for purposes of implementing wz_mini still possible via the factory reset process, or does the inherent encryption mentioned also prohibit the implementation of wz_mini in rolled back firmware?

**Comment by endertable** [comment]
Good question.  The only way they can prevent wyze_mini from functioning at all is by altering the Uboot, which I don’t think they have.  What they did do is prevent from hooking onto the video streams by removing some of the dynamic libraries.  So homebrew RTSP and the “cmd” features are gone on the new FW, for now.  Factory reset to prior firmware is still possible as well. But you may have to go through the full set up process on the app.

**Comment by sideup66** [comment]
Does anyone know if the dnsmasq fixes are still working on the new firmware? Id imagine so...my own setup doesnt rely on RTSP so that change doesnt really affect me. I just need cron plus the dns fixes....

**Comment by skilau** [comment]
I guess on a somewhat ironic note, their upgrade appears to be bricking some units, so they may end up having to pull it back.
https://forums.wyze.com/t/wyze-cam-v3-stuck-on-solid-red-light/186720/321

**Comment by skilau** [comment]
I went and force upgraded one of my spare v3's to the new firmware, just so I can see how bad things are.

On the positive side, we are still able to use Ethernet, and the Ethernet/Wifi bonding, which is good.

I will dig in more tonight.

**Reply by sideup66** [reply]
I think anything NOT related to RTSP should work....we have root on the box afterall and the boot process is literally an exploit existing in the camera's hardware. can you confirm on the dns front if the fixes there still hold? ie the dnsmasq server is still functional? at the moment im freezing mine on the previous build since it + hacks seem to have it in a good place stability wise.

**Comment by Bob-AOL** [comment]
I still don't understand the bonding unless used for failover from wifi to wired which would be a very unlikely configuration in the first place if ethernet/POE were available. Failover from ethernet to wifi makes even less sense due to the very high probability that if the ethernet connection has failed there would also be no power to the device (if powered over the same ethernet connection., of course).

But as usual, I could be overlooking a very obvious reason for having such a feature available -- even if it's simply because it's possible.

**Reply by skilau** [reply]
I have my cams up pretty high up, and I live in Minnesota.

During the winter, if somehow the wired connection fails (no matter how unlikely), I want to have a backup chance at having the camera still work, because getting the ladder out when it is -20F out, and a foot of snow outside, is pretty dicey.

But you are correct, that if the Ethernet is dead, it is likely the camera is dead too.

**Comment by Bob-AOL** [comment]
So can the new firmware be rolled back to 4.36.9.139  to facilitate wz_mini without major issues and hoop jumping? Not much follow-up on the subject.

**Reply by sideup66** [reply]
You can roll back a v3 to 4.36.139 to use wz_mini just fine. Only obstacle really is that your camera will need to be reset after the rollback completes. This is because the new version encrypts the wifi conf file containing the accesspoint name and password used to connect.

**Reply by Bob-AOL** [reply]
Thx. Wasn't quite sure about the actual context of the wifi.conf encryption issue. Home Depot has a two-for-one V3 sale going and I was hoping to snag a couple if wz_mini still flies without too much hassle. I'm on my way!

**Reply by skilau** [reply]
Yup, works fine to roll back. 

PS: I suspect they are having issues with their "new" firmware, and have stopped rolling it out.

The new cams I got from the same Home Depot 2 for 1 deal came with even older firmware, and when it asked me to update.
It updated to our "good" 4.36.9.139 firmware, and not the latest one that breaks us.

**Reply by Bob-AOL** [reply]
Just got home with mine. These had 4.36.8.15 and also upgraded to 4.36.9.139.

**Reply by youmansk** [reply]
ok. i went to 4.36.9.139. booted with the sd card and files on it. my red light has a slow blink and the ir led flash every second or so. What am i missing? I have fat32 on a 32g sd card.
